### PR TITLE
Cigar operation

### PR DESCRIPTION
**Pull request by marehr**
Resolves https://github.com/seqan/product_backlog/issues/294.

### CODE REVIEW

File: include/seqan3/alphabet/cigar/exposition_only/cigar_operation.hpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+/*!\brief The actual implementation of seqan3::cigar::operation for documentation purpose-only.
+ * \implements seqan3::writable_alphabet
+ * \if DEV \implements seqan3::detail::writable_constexpr_alphabet \endif
+ * \implements seqan3::trivially_copyable
+ * \implements seqan3::standard_layout
+ * \implements std::regular
+ *
+ * \details
+ *
+ * \note Please note that this class only exists, because of technically reasons.
```
Comment (eseiler) [inline include/seqan3/alphabet/cigar/exposition_only/cigar_operation.hpp:33]:
```suggestion
 * \note Please note that this class only exists because of technically reasons.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -88,8 +88,13 @@ If possible, provide tooling that performs the changes, e.g. a shell-script.
 
 #### Alphabet
 
-* Removed seqan3::char_is_valid_for requirement from seqan3::writable_alphabet and seqan3::writable_constexpr_alphabet
+* Removed seqan3::char_is_valid_for requirement from seqan3::detail::writable_alphabet and
```
Comment (MitraDarja) [inline CHANGELOG.md:91]: This change looks like it would belong to a different PR? Or did you just fix this because you touched the file already?

**Review by MitraDarja — CHANGES_REQUESTED** [review]
One really minor thing, otherwise it lgtm. :)

File: CHANGELOG.md (hunk 2 above)
Comment (marehr) [inline CHANGELOG.md:91]:
```suggestion
* Removed seqan3::char_is_valid_for requirement from seqan3::writable_alphabet and
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -88,8 +88,13 @@ If possible, provide tooling that performs the changes, e.g. a shell-script.
...
+  seqan3::writable_constexpr_alphabet
```
Comment (marehr) [inline CHANGELOG.md:92]:
```suggestion
  seqan3::detail::writable_constexpr_alphabet
```

File: CHANGELOG.md (hunk 2 above)
Comment (MitraDarja) [reply CHANGELOG.md]: No, that is not necessary. Just wanted to make sure, that this was on purpose and not an accident. ;) 
But, this is still shown as a change? (I think because there is a new line that is not in the master branch?)

File: CHANGELOG.md [hunk 4]
```diff
@@ -88,8 +88,13 @@ If possible, provide tooling that performs the changes, e.g. a shell-script.
 
 #### Alphabet
 
-* Removed seqan3::char_is_valid_for requirement from seqan3::writable_alphabet and seqan3::writable_constexpr_alphabet
+* Removed seqan3::char_is_valid_for requirement from seqan3::writable_alphabet and
```
Comment (MitraDarja) [inline CHANGELOG.md:91]: This is still a change due to the new line after and. Maybe change that back? :)

**Review by MitraDarja — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (marehr) [reply CHANGELOG.md]: I added `detail::` and needed to wrap the line, as it exceeded our line limit :)

File: CHANGELOG.md (hunk 4 above)
Comment (marehr) [reply CHANGELOG.md:91]: I added `detail::` and needed to wrap the line, as it exceeded our line limit :)

**Review by MitraDarja — APPROVED** [review]

**Review by eseiler — APPROVED** [review]
